front/src/AppNavbar.tsx

```diff
@@ -35,34 +35,39 @@ export function AppNavbar() {
         <MainNavbar>
           <NavItem
             label="Search"
-            icon={<IconSearch size={theme.icon.size.md} />}
+            Icon={IconSearch}
+            iconProps={{ size: theme.icon.size.md }}
```

Comment:
We shouldn't introduce an iconProps for now because NavItem is responsible for setting the size of its icons. 

You can consider that every Icon that is using a size, this size should be set by the component which receives the icon.